bundles/com.salesforce.bazel.eclipse.core/src/com/salesforce/bazel/eclipse/core/model/discovery/BaseProvisioningStrategy.java

```diff
@@ -221,6 +221,304 @@ static IMarker createMarker(IResource resource, String type, IStatus status) thr
 
     private String jreSystemLibraryConfiguration;
 
+    /**
+     * Set to track temporary placeholder Java files created during provisioning
+     */
+    private final Set<java.nio.file.Path> temporaryPlaceholderFiles = new HashSet<>();
+
+    /**
+     * Ensures that at least one Java file exists in the package directory when srcs contains content. If no Java files
+     * exist in the BUILD file's directory, creates a temporary placeholder Java file.
+     *
+     * @param bazelTarget
+     *            the Bazel target to check
+     * @param srcs
+     *            the list of source files from the target's srcs attribute
+     * @return the temporary file name if created, or null if no temporary file was needed
+     * @throws CoreException
+     */
+    private String ensureJavaFileExistsInPackage(BazelTarget bazelTarget, List<String> srcs) throws CoreException {
```

Comment:
Why does no java file exists? Is it generated by some other target and copied into the source tree?

Reply:
Right but it does not belong here.

The logic to detect source directories is here:
https://github.com/eclipseguru/bazel-eclipse/blob/9542833a53db76116f19376df94178f0c8b9513f/bundles/com.salesforce.bazel.eclipse.core/src/com/salesforce/bazel/eclipse/core/model/discovery/projects/JavaSourceInfo.java#L114

It looks like it's missing to identify that particular setup.